libsolidity/formal/CHC.cpp

```diff
@@ -642,10 +642,18 @@ void CHC::externalFunctionCall(FunctionCall const& _funCall)
 	}
 
 	auto postCallState = vector<smtutil::Expression>{state().state()} + currentStateVariables();
-	auto nondet = (*m_nondetInterfaces.at(m_currentContract))(preCallState + postCallState);
+	auto error = errorFlag().increaseIndex();
+	auto nondet = (*m_nondetInterfaces.at(m_currentContract))(preCallState + postCallState + vector<smtutil::Expression>{error, state().thisAddress()});
 	// TODO this could instead add the summary of the called function, where that summary
 	// basically has the nondet interface of this summary as a constraint.
 	m_context.addAssertion(nondet);
+	solAssert(m_errorDest, "");
+	connectBlocks(m_currentBlock, predicate(*m_errorDest), errorFlag().currentValue() > 0);
+	solAssert(m_currentContract && m_currentFunction, "");
+	for (auto const* base: m_currentContract->annotation().linearizedBaseContracts)
+		for (auto const* defined_function: base->definedFunctions())
+			if (!defined_function->isConstructor() && defined_function->isPublic())
+				m_callGraph[m_currentFunction].insert(defined_function);
```

Comment:
Probably add a comment that here we record that all external methods of the contract could be called from the external call made by us.

Reply:
Comment added.